_overviews/tutorials/scala-for-java-programmers.md

```diff
@@ -159,85 +220,133 @@ types.
+This also means that `+`, `*`, etc. are valid identifiers for fields/methods/etc
 in Scala.
 
 ### Functions are objects
 
-Functions are also
-objects in Scala. It is therefore possible to pass functions as
-arguments, to store them in variables, and to return them from other
-functions. This ability to manipulate functions as values is one of
-the cornerstone of a very interesting programming paradigm called
-*functional programming*.
-
-As a very simple example of why it can be useful to use functions as
-values, let's consider a timer function whose aim is to perform some
-action every second. How do we pass it the action to perform? Quite
-logically, as a function. This very simple kind of function passing
-should be familiar to many programmers: it is often used in
-user-interface code, to register call-back functions which get called
-when some event occurs.
+True to _everything_ being an object, in Scala even functions are objects, going beyond Java's support for
+lambda expressions.
+
+Compared to Java, there is very little difference between function objects and methods: you can pass methods as
+arguments, store them in variables, and return them from other functions, all without special syntax.
+This ability to manipulate functions as values is one of the cornerstones of a very
+interesting programming paradigm called *functional programming*.
+
+To demonstrate, consider a timer function which
+performs some action every second. The action to be performed is supplied by the
+caller as a function value.
 
 In the following program, the timer function is called
 `oncePerSecond`, and it gets a call-back function as argument.
 The type of this function is written `() => Unit` and is the type
-of all functions which take no arguments and return nothing (the type
-`Unit` is similar to `void` in C/C++). The main function of
-this program simply calls this timer function with a call-back which
-prints a sentence on the terminal. In other words, this program
-endlessly prints the sentence "time flies like an arrow" every
+of all functions which take no arguments and return no useful value
+(as before, the type `Unit` is similar to `void` in Java).
+
+The entry-point of this program calls `oncePerSecond` by directly passing
+the `timeFlies` method.
+
+In the end this program will infitely print the sentence `time flies like an arrow` every
 second.
 
-    object Timer {
-      def oncePerSecond(callback: () => Unit): Unit = {
-        while (true) { callback(); Thread sleep 1000 }
-      }
-      def timeFlies(): Unit = {
-        println("time flies like an arrow...")
-      }
-      def main(args: Array[String]): Unit = {
-        oncePerSecond(timeFlies)
-      }
-    }
+{% tabs callback-demo class=tabs-scala-version %}
+
+{% tab 'Scala 2' for=callback-demo %}
+```scala
+object Timer {
+  def oncePerSecond(callback: () => Unit): Unit = {
+    while (true) { callback(); Thread sleep 1000 }
```

Comment:
I think there is an opportunity to change the code style to this:

```suggestion
    while (true) { callback(); Thread.sleep(1000) }
```

Reply:
done